test/e2e/federated-service.go

```diff
@@ -285,16 +299,52 @@ func discoverService(f *framework.Framework, name string, exists bool) {
 					Command: command,
 				},
 			},
-			RestartPolicy: api.RestartPolicyNever,
+			RestartPolicy: api.RestartPolicyOnFailure,
 		},
 	}
-	if exists {
-		f.TestContainerOutputRegexp("federated service discovery", pod, 0, []string{
-			`Name:\s+` + FederatedDNS1123Regexp + `\nAddress \d+:\s+` + FederatedIPAddrRegexp + `\s+` + FederatedDNS1123Regexp,
-		})
-	} else {
-		f.TestContainerOutputRegexp("federated service discovery", pod, 0, []string{
-			`nslookup: can't resolve '` + name + `'`,
-		})
+
+	_, err := f.Client.Pods(f.Namespace.Name).Create(pod)
+	Expect(err).
+		NotTo(HaveOccurred(), "Trying to create pod to run %q", command)
+
+	// If we ever get any container logs, stash them here.
+	logs := ""
+
+	logerr := func(err error) error {
```

Comment:
Perhaps neater and more re-usable to factor this out into a separate function that others can use?